tests/migrations/test_runner_individual.py

```diff
@@ -0,0 +1,235 @@
+            .get_processor()
+            .process_message(
+                (2, "insert", raw_transaction),
+                KafkaMessageMetadata(0, 0, datetime.utcnow()),
+            )
+        )
+        rows.extend(processed.rows)
+
+    BatchWriterEncoderWrapper(
+        table_writer.get_batch_writer(metrics=DummyMetricsBackend(strict=True)),
+        JSONRowEncoder(),
+    ).write(rows)
+
+
+def test_groupedmessages_compatibility() -> None:
+
+    # these three commands are repeated
+    cluster = get_cluster(StorageSetKey.EVENTS)
+    database = cluster.get_database()
+    connection = cluster.get_query_connection(ClickhouseClientSettings.MIGRATE)
+
+    # Create old style table witihout project ID
+    connection.execute(
+        """
+        CREATE TABLE groupedmessage_local (`offset` UInt64, `record_deleted` UInt8,
+        `id` UInt64, `status` Nullable(UInt8), `last_seen` Nullable(DateTime),
+        `first_seen` Nullable(DateTime), `active_at` Nullable(DateTime),
+        `first_release_id` Nullable(UInt64)) ENGINE = ReplacingMergeTree(offset)
+        ORDER BY id SAMPLE BY id SETTINGS index_granularity = 8192
+        """
+    )
+
+    migration_id = "0010_groupedmessages_onpremise_compatibility"
+
+    # Runner() instantiation, run_migration(), update status
+    runner = Runner()
+    runner.run_migration(MigrationKey(MigrationGroup.SYSTEM, "0001_migrations"))
+    events_migrations = get_group_loader(MigrationGroup.EVENTS).get_migrations()
+
+    # Mark prior migrations complete
+    for migration in events_migrations[: (events_migrations.index(migration_id))]:
+        runner._update_migration_status(
+            MigrationKey(MigrationGroup.EVENTS, migration), Status.COMPLETED
+        )
+
+    runner.run_migration(
+        MigrationKey(MigrationGroup.EVENTS, migration_id), force=True,
+    )
+
+    assert connection.execute(
+        f"SELECT primary_key FROM system.tables WHERE name = 'groupedmessage_local' AND database = '{database}'"
+    ) == [("project_id, id",)]
+
+
+# provide a migration ID, group and run all the migrations that come before
+def run_prior_migrations(migration_group, stop_migration_id, runner):
+
+    right_migrations = next(
+        group_migrations
+        for (group, group_migrations) in runner.show_all()
+        if group == migration_group
+    )
+
+    # Run migrations up to the desired 'stop' ID
+    for migration in right_migrations:
+        if migration.migration_id == stop_migration_id:
+            break
+
+        runner.run_migration(
+            MigrationKey(migration_group, migration.migration_id), force=True
+        )
+
+
+# Below is an example of a valid test
+# Write a similar test case potentially ?
+def test_backfill_errors() -> None:
+
+    backfill_migration_id = "0014_backfill_errors"
+    runner = Runner()
+    runner.run_migration(MigrationKey(MigrationGroup.SYSTEM, "0001_migrations"))
+
+    run_prior_migrations(MigrationGroup.EVENTS, backfill_migration_id, runner)
```

Comment:
This method is only once. If you do not plan to reuse in other circumstances, likely you do not need to split it out of the main method.

Reply:
I think we may like to keep this `run_prior_migrations` function as it will be useful in writing future tests for individual migrations (Lyn and I discussed).